txnbuild/transaction.go

```diff
@@ -26,38 +27,195 @@ import (
 	"github.com/stellar/go/xdr"
 )
 
+// MinBaseFee is the minimum base for the Stellar network
+const MinBaseFee = 100
+
 // Account represents the aspects of a Stellar account necessary to construct transactions. See
 // https://www.stellar.org/developers/guides/concepts/accounts.html
 type Account interface {
 	GetAccountID() string
 	IncrementSequenceNumber() (xdr.SequenceNumber, error)
-	// Action needed in release: horizonclient-v2.0.0
-	// add GetSequenceNumber method
-	// GetSequenceNumber() (xdr.SequenceNumber, error)
+	GetSequenceNumber() (xdr.SequenceNumber, error)
 }
 
-// Transaction represents a Stellar transaction. See
-// https://www.stellar.org/developers/guides/concepts/transactions.html
 type Transaction struct {
-	SourceAccount  Account
-	Operations     []Operation
-	BaseFee        uint32
-	Memo           Memo
-	Timebounds     Timebounds
-	Network        string
-	xdrTransaction xdr.TransactionV0
-	xdrEnvelope    *xdr.TransactionV0Envelope
+	envelope      xdr.TransactionEnvelope
+	baseFee       int64
+	totalFee      int64
+	isFeeBump     bool
+	feeSource     string
+	innerTx       *Transaction
+	sourceAccount SimpleAccount
+	operations    []Operation
+	memo          Memo
+	timebounds    Timebounds
+	signatures    []xdr.DecoratedSignature
+}
+
+func (t *Transaction) BaseFee() int64 {
+	return t.baseFee
+}
+
+func (t *Transaction) TotalFee() int64 {
```

Comment:
Don't have a strong opinion but maybe it's worth keeping the name consistent with Horizon.
```suggestion
func (t *Transaction) MaxFee() int64 {
```